src/components/Accounts/Actions/Actions.js

```diff
@@ -438,9 +443,15 @@ class Actions extends React.Component {
       type.status.name = 'Closed';
     }
     const balance = type.remaining - parseFloat(waive);
-    const comment = values.comment ? (tagStaff + ': ' + values.comment) : '';
```

Comment:
As I see [transfer](https://github.com/folio-org/ui-users/pull/923/files#diff-556cff218755ef7b8f01dba9665e8432R366) method has a lot of duplication stuff as [waive](https://github.com/folio-org/ui-users/pull/923/files#diff-556cff218755ef7b8f01dba9665e8432L428). That is why `commentInfo` stuff is duplicated in both of them. Please come with code without duplication logic in two places.

Reply:
We add the ``validateComment()`` function, to reduce the duplicate code of ``pay``, ``waive`` and ``transfer`` relative to ``commentInfo``. Does this solve the suggestion you made to us?

Reply:
I suggest using the following approach to get rid of duplicated logic.

```
  transfer = (type, amount, values) => {
    this.props.mutator.activeRecord.update({ id: type.id });

    const paymentMeta = this.preparePaymentMeta({
      paymentMethod: values.account,
      action: 'transfer',
      payment: amount,
      type,
    });

    return this.editAccount(type, paymentMeta.paymentStatus, type.status.name, paymentMeta.balance)
      .then(() => this.newAction(paymentMeta.action, type.id, paymentMeta.paymentStatus, amount, this.validateComment(values), paymentMeta.balance, 0, type.feeFineOwner));
  }

  pay = (type, amount, values) => {
    this.props.mutator.activeRecord.update({ id: type.id });

    const paymentMeta = this.preparePaymentMeta({
      paymentMethod: values.method,
      action: 'pay',
      payment: amount,
      type,
    });

    const owners = _.get(this.props.resources, ['owners', 'records'], []);
    const createdAt = (owners.find(o => o.id === values.ownerId) || {}).owner;
    return this.editAccount(type, paymentMeta.paymentStatus, type.status.name, paymentMeta.balance)
      .then(() => this.newAction(paymentMeta.action, type.id, paymentMeta.paymentStatus, amount, this.validateComment(values), paymentMeta.balance, values.transaction, createdAt || type.feeFineOwner));
  }

  preparePaymentMeta = ({ paymentMethod, payment, type, action }) => {
    const isFullPayment = payment >= type.remaining;

    this.markPaymentAsClosed(type, isFullPayment);

    return {
      action: { paymentMethod },
      paymentStatus: this.getPaymentStatusMessage(action, isFullPayment),
      balance: type.remaining - parseFloat(payment);
    }
  };

  markPaymentAsClosed = (type, isFullPayment) => {
    if (isFullPayment) {
      type.status.name = 'Closed';
    }
  }

  getPaymentStatusMessage = (action, isFullPayment) => {
    const { intl: { formatMessage } } = this.props;
    const paymentStatusBeginning = _.capitalize(formatMessage({ id: `ui-users.accounts.actions.warning.${action}Action` }));

    return isFullPayment
     ? `${paymentStatusBeginning} ${formatMessage({ id: 'ui-users.accounts.status.fully' })}
     : `${paymentStatusBeginning} ${formatMessage({ id: 'ui-users.accounts.status.partially' })}`
  }
```

Reply:
SonarQube is not reporting duplicity, so what would be the purpose of your suggestion?

Reply:
Even though Sonar does not if you compare both methods they look almost the same.